src/metatrain/utils/io.py

```diff
@@ -99,15 +101,62 @@ def load_model(
         )
 
     if Path(path).suffix in [".yaml", ".yml"]:
-        raise ValueError(f"path '{path}' seems to be a YAML option file and no model")
+        raise ValueError(
+            f"path '{path}' seems to be a YAML option file and not a model"
+        )
 
-    if urlparse(str(path)).scheme:
+    # Download from HuggingFace with a private token
+    if kwargs.get("huggingface_api_token"):
+        try:
+            from huggingface_hub import hf_hub_download
+        except ImportError:
+            raise ImportError(
+                "To download a model from HuggingFace, please install the "
+                "`huggingface_hub` package with pip (`pip install "
+                "huggingface_hub`)."
+            )
+        path = str(path)
+        if not path.startswith("https://huggingface.co/"):
+            raise ValueError(
+                f"Invalid URL '{path}'. HuggingFace models should start with "
+                "'https://huggingface.co/'."
+            )
+        # get repo_id and filename
+        split_path = path.split("/")
+        repo_id = f"{split_path[3]}/{split_path[4]}"  # org/repo
+        filename = ""
+        for i in range(5, len(split_path)):
+            filename += split_path[i] + "/"
+        filename = filename[:-1]
+        if filename.startswith("resolve"):
+            if not filename[8:].startswith("main/"):
+                raise ValueError(
+                    f"Invalid URL '{path}'. metatrain only supports models from the "
+                    "'main' branch."
+                )
+            filename = filename[13:]
+        if filename.startswith("blob/"):
+            if not filename[5:].startswith("main/"):
+                raise ValueError(
+                    f"Invalid URL '{path}'. metatrain only supports models from the "
+                    "'main' branch."
+                )
+            filename = filename[10:]
+        path = hf_hub_download(repo_id, filename, token=kwargs["huggingface_api_token"])
+        # make sure to copy the checkpoint to the current directory
+        shutil.copy(path, Path.cwd() / str(path).split("/")[-1])
```

Comment:
Should this be added to the log? Some kind of `INFO: downloaded $FILE_NAME from hugingface.co/owner/repo to current directory`

Reply:
Sure